qiskit_nature/problems/second_quantization/lattice/models/ising_model.py

```diff
@@ -0,0 +1,120 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""The Ising model"""
+import numpy as np
+from retworkx import PyGraph
+
+from qiskit_nature.operators.second_quantization import SpinOp
+from qiskit_nature.problems.second_quantization.lattice.lattices import Lattice
+
+
+class IsingModel:
+    """The Ising model."""
+
+    def __init__(self, lattice: Lattice) -> None:
+        """
+        Args:
+            lattice: Lattice on which the model is defined.
+        """
+        self._lattice = lattice
+
+    @property
+    def lattice(self) -> Lattice:
+        """Return a copy of the input lattice."""
+        return self._lattice.copy()
```

Comment:
Since this is identical to the code in `FermiHubbardModel`, it would make sense to extract common base class for these kind of models.